apps/src/openni_mls_smoothing.cpp

```diff
@@ -203,17 +203,15 @@ usage(char** argv)
 int
 main(int argc, char** argv)
 {
-  if (argc < 2) {
+  if (pcl::console::find_argument(argc, argv, "-h") != -1 ||
+      pcl::console::find_argument(argc, argv, "--help") != -1) {
     usage(argv);
     return 1;
   }
 
-  std::string arg(argv[1]);
-
-  if (arg == "--help" || arg == "-h") {
-    usage(argv);
-    return 1;
-  }
+  std::string device_id = "";
+  if (argc > 1 && argv[1][0] != '-')
```

Comment:
```suggestion
  if (argc > 1)
```
Can be removed now that we explicitly checks for -h / -help ?

Reply:
There are more cases than those I highlighted.

Maybe make the device-id an argument as well, so it can be parsed with:

std::string device_id;
pcl::console::parse_argument(argc, argv, "-device_id", device_id);

Then it doesn't have to be first and its more obvious its optional?

Reply:
Hi! I kept it in because one can still set the options using - 

for mls_smoothing it would look something like ./argv[0] -search_radius 10

if it's not secured with - then it would take -search_radius as the device_id, no?

but you are right! I think its better if device_id is also parsed individually as well as being independent of its position, ill work on this for all the affected files as well as improve the usage() function :)